newsfragments/1457.feature.rst

```diff
@@ -0,0 +1,7 @@
+When exiting a nursery block, the parent task always waits for child
+tasks to exit. This wait cannot be cancelled. However, previously, if
+you tried to cancel it, it *would* inject a `Cancelled` exception,
+even though it wasn't cancelled. Most users probably never noticed
```

Comment:
> This wait cannot be cancelled. However, previously, if
you tried to cancel it, it would inject a Cancelled exception,
even though it wasn't cancelled.

So this means the `Cancelled` is still injected but now is ignored via the new [`isinstance` check](https://github.com/python-trio/trio/pull/1696/files#diff-568b66d9830d9516b524413eb041a1d4c8ccf0b76af4e10d96098888ce323235R924)?

Also if the following is true:

> This wait cannot be cancelled. However, previously, if
you tried to cancel it, it would inject a Cancelled exception,
even though it wasn't cancelled.

Then why [the change to ` cancel_shielded_checkpoint()`](https://github.com/python-trio/trio/pull/1696/files#diff-568b66d9830d9516b524413eb041a1d4c8ccf0b76af4e10d96098888ce323235R935)?